src/components/Navigation/NavBar.js

```diff
@@ -9,11 +9,34 @@ import Glyphicon from 'react-bootstrap/lib/Glyphicon';
 import Navbar from 'react-bootstrap/lib/Navbar';
 import FormGroup from 'react-bootstrap/lib/FormGroup';
 import FormControl from 'react-bootstrap/lib/FormControl';
-
+import MenuItem from 'react-bootstrap/lib/MenuItem';
+import Dropdown from 'react-bootstrap/lib/Dropdown';
 import NavLink from './NavLink';
-import FlagGroup from './FlagGroup';
 
 export const NavBar = React.createClass({
+
+  getInitialState() {
+    return {
+      selectedLanguage: 'Norsk'
+    };
+  },
+
+  handle(eventKey) {
+    switch (eventKey) {
+      case 'norway':
+        this.props.setLanguageNorway();
+        this.setState({selectedLanguage: 'Norsk'});
+        break;
+      case 'sweden':
+        this.props.setLanguageSweden();
+        this.setState({selectedLanguage: 'Svensk'});
+        break;
+      case 'denmark':
+        this.props.setLanguageDenmark();
+        this.setState({selectedLanguage: 'Dansk'});
```

Comment:
Intern state tar bare være på språket som vises for brukeren på knappen. Altså, det språket som er valgt og gjeldende. Eller er det greit å vise 'Norway', 'Sweden' eller 'Denmark' på knappen?